language-support/ts/defs.bzl

```diff
@@ -1,21 +1,24 @@
 # Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-load("@language_support_ts_deps//typescript:index.bzl", "tsc")
+load("@build_bazel_rules_nodejs//:providers.bzl", "DeclarationInfo", "LinkablePackageInfo")
+load("@npm_bazel_typescript//:index.bzl", "ts_project")
 
 def _da_ts_library_impl(ctx):
+    path = "/".join([p for p in [ctx.bin_dir.path, ctx.label.workspace_root, ctx.label.package] if p])
+    print(path)
+    print(ctx.attr.module_name)
     return [
-        DefaultInfo(
-            files = depset(ctx.files.srcs),
-            runfiles = ctx.runfiles(files = ctx.files.srcs),
-        ),
-    ]
+        ctx.attr.dep[DefaultInfo],
+        ctx.attr.dep[DeclarationInfo],
+        LinkablePackageInfo(package_name = ctx.attr.module_name, path = path)]
 
+# TODO This doesn’t quite work, we need a DeclarationInfo provider.
+# ts_project has this but it doesn’t have module_name and module_root.
```

Comment:
How does it fail?

The documentation to `ts_project` states that one should [define `rootDirs` in `tsconfig.json`](https://github.com/bazelbuild/rules_nodejs/blob/bd83907954d1b3cd53dff01c37e3c4964890ef5c/packages/typescript/src/internal/ts_project.bzl#L240-L262), have you tried that?

Reply:
It fails when building `daml-ledger` with
```
language-support/ts/daml-ledger/index.test.ts(4,46): error TS2307: Cannot find module '@daml/types'.
```
I did try setting `rootDirs` (just pushed my attempt) but that doesn’t seem to help. The docs also say that this is only relevant for relative imports, not quite sure if that applies here or not.